fhir-server/src/main/java/com/ibm/fhir/server/resources/FHIRResource.java

```diff
@@ -419,12 +422,17 @@ protected String getRequestBaseUri(String type) throws Exception {
 
         // Strip off any path elements after the base
         if (type != null && !type.isEmpty()) {
-            int resourceNamePathLocation = baseUri.lastIndexOf("/" + type);
+            int resourceNamePathLocation = baseUri.indexOf("/" + type + "/");
             if (resourceNamePathLocation != -1) {
```

Comment:
In this case, where you've found `/Patient/`, maybe you want to do an `endsWith("/" + type)` check and bump out to that index if found. Would correctly parse this path: `https://example.com/my/Patient/api/Patient`. Guess you'd want to check that it didn't directly follow a `/Patient` to make sure it wasn't a logical ID (i.e. `https://example.com/my/Patient/api/Patient/Patient`)

Reply:
sorry, i'm not sure I understand the suggestion

Reply:
This is a moot point if we're assuming no path segment matching resource type exactly and check in line 425 doesn't change to use `lastIndexOf`